File: 002/haskell/002.hs
-- Project Euler Problem # 2
-- Even Fibonacci Numberss
-- Answer:
--
-- Created by Thomas Nelson <tn90ca@gmail.com>
-- Created on 2015-01-17

module Main where

-- Prints the nth fibonacci number
slowfib 0 = 1
slowfib 1 = 1
slowfib n = (slowfib (n - 1)) + (slowfib (n - 2))

fib = 1:1:zipWith (+) fib (tail fib)

terms  = take 33 fib
nterms = filter (\x -> mod x 2 == 0 && x < 4000000)terms

main = print (sum $ nterms)

File: 002/c++/002.cpp
// Project Euler Problem # 2
// Even Fibonacci Numbers
// Answer: 4613732
//
// Created by Thomas Nelson <tn90ca@gmail.com>
// Created on 2015-05-13

#include <iostream>

int main () {
    int sum = 0;
    int a = 1;
    int b = 1;

    while (a <= 4000000 && b <= 4000000) {
        int c = a + b;

        if (c%2 == 0) {
            sum += c;
        }

        a = b;
        b = c;
    }
    
    std::cout << "Sum = " << sum << '\n';

    return 0;
}